protocol/network/6-interfaces.md

```diff
@@ -0,0 +1,137 @@
+6 Interfaces
+============
+
+## 6.1 libp2p
+
+## 6.2 Peer Routing
+
+## 6.3 Swarm
+
+~~The network is abstracted through the swarm which presents a simplified interface for the remaining layers to have access to the network. This interface should look like:~~
+
+- `.openStream(peer, protocol)` - peer should contain the ID of the peer and its respective multiaddrs known.
+- `.registerHandler(protocol, handlerFunc)` - enable a protocol to be registered, so that another peer can open a stream to talk with us to that specific protocol
+- `.listen()` - to start listening for incoming connections and therefore opening of streams
```

Comment:
The interface has evolved a lil bit with the last 'Swarm revisit' to support multiprotocols, but it is essentially the same:

https://github.com/diasdavid/node-libp2p-swarm#usage

> we want users to use a libp2p.Node, not the Swarm directly.
Totally agree, we can have a 'front facing' interface from libp2p, in fact node-libp2p is just that, glue, take a peak at: 
https://github.com/diasdavid/node-libp2p#setting-everything-up

Glad to remove the prefixes and make it everything libp2p.<call>